### PR TITLE
tests/fde-on-classic: ask for generic serial to the store

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
Store should now provide these assertions and we set
"serial-authority: generic" in the assertion for the test already.

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you

**Comment by mvo5** [comment]
I pushed a tiny fix to wait for the init but it still fails with:
```
2022-11-22T11:39:06Z ERROR cannot deliver device serial request: invalid assertion: could not validate model assertion (no matching public key "EAD4DbLxK_kn0gzNCXOs3kd6DeMU3f-L6BEsSEuJGBqCORR0gXkdDxMbOm11mRFu" for signature by "developer1")
```

File: tests/nested/manual/fakeinstaller/task.yaml [hunk 1]
```diff
@@ -172,6 +171,12 @@ execute: |
   remote.exec "snap changes" | MATCH "Done.* Initialize system state"
   remote.exec "snap list" | MATCH pc-kernel
 
+  # Check for the generic serial assertion
+  retry -n 100 sh -c 'remote.exec "snap changes" | MATCH "Done.*Initialize device"'
+  remote.exec "snap model --serial --assertion" | MATCH "authority-id: generic"
+  remote.exec "snap model --serial --assertion" | MATCH "brand-id: generic"
+  remote.exec "snap model --serial --assertion" | MATCH "model: generic-classic"
```
Comment (pedronis) [inline tests/nested/manual/fakeinstaller/task.yaml:178]: these two (brand-id, model) should be the values from the model used, not the ones for the generic assertion shipped with snapd

**Review by pedronis — CHANGES_REQUESTED** [review]
the checks are not quite right, assuming the store works

File: tests/nested/manual/fakeinstaller/task.yaml (hunk 1 above)
Comment (alfonsosanchezbeato) [reply tests/nested/manual/fakeinstaller/task.yaml:178]: Fixed

**Comment by alfonsosanchezbeato** [comment]
FTR I have not seen the store returning a valid serial for a classic model that I signed with my store key and that finished seeding successfully. Error returned is:
```
ERROR cannot deliver device serial request: Cannot process serial request for device with brand "tcMZ22pMaY5EVwoLozfjM4fR31bko4yj" and model "xinyi-classic-22-amd64"
```
So I don't think this is supported yet.

**Comment by pedronis** [comment]
do you have serial-authority: generic in your model?

**Review by pedronis — APPROVED** [review]
this looks good and is passing now afaict, but the summary is wrong as the changed test is not fakeinstaller?

**Comment by alfonsosanchezbeato** [comment]
> this looks good and is passing now afaict, but the summary is wrong as the changed test is not fakeinstaller?

Right, changed now

**Pull request merged**